public/js/ClusterPlot.js

```diff
@@ -0,0 +1,212 @@
+var d3 = require('d3');
+require('d3-tip')(d3); // add d3-tip plugin to d3 namespace
+
+/**
+ * ClusterPlot: Object to draw scatterplot of dimension reduced data.
+ *
+ * This scatterplot is controlled by a slider that allows the user to
+ * select the number of k partitions they would like the data clustered into.
+ * The plot points are coloured according to which cluster they belong to for
+ * the currently selected k.
+ *
+ * @constructor
+ * @param {array} sampleData - The sample data for the screen. Each element
+ * in the array is an instance of a Sample document.
```

Comment:
Can you indent this four spaces for better visual parsing of all the parameters? e.g.:

```
 * @param {array} sampleData - The sample data for the screen. Each element
 *     in the array is an instance of a Sample document.
 * @param {string} element - The ID of the target div for this plot.
```